tests/fsharp/core/fsfromfsviacs/lib2.cs

```diff
@@ -60,3 +60,14 @@ internal class SomeClass
 
     }
 }
+
+namespace FSharpOptionalTests
+{
+    public class ApiWrapper 
+    {
+        public static Tuple<FSharpOption<T>, FSharpOption<int>> MethodWithOptionalParams<T>(T value1, int value2)
+        {
+            return OptionalParameterTests.API.MethodWithOptionalParams<T>(value1 = value1, value2 = value2);
```

Comment:
Please invoke this method from the actual test.fs?  

Also please add further tests for other uses of the implicit conversion (ones that just utilize the conversion independent of optional parameters).